Source/Core/FFmpeg_Glue.h

```diff
@@ -69,9 +72,126 @@ class FFmpeg_Glue
+
+        // Cache
+        std::vector<AVFrame*>*  FramesCache;
+        AVFrame*                FramesCache_Default;
+    };
+    struct outputdata
+    {
+        // Constructor / Destructor
+        outputdata(int index);
+        ~outputdata();
+
+        int index = {0};
+
+        //Actions
+        void                    Process(AVFrame* DecodedFrame);
+        void                    ApplyFilter(const AVFramePtr& sourceFrame);
+        void                    ApplyScale(const AVFramePtr& sourceFrame);
+        void                    ReplaceImage();
+        void                    AddThumbnail();
+        void                    AddPanel();
+
+        // In
+        bool                    Enabled;
+        string                  Filter;
+        string                  Title;
+
+        // FFmpeg pointers - Input
+        int                     Type;
+        AVStream*               Stream;
+        AVFramePtr              DecodedFrame;
+
+        // FFmpeg pointers - Filter
+        AVFilterGraph*          FilterGraph;
+        AVFilterContext*        FilterGraph_Source_Context;
+        AVFilterContext*        FilterGraph_Sink_Context;
+        AVFramePtr              FilteredFrame;
+
+        // FFmpeg pointers - Scale
+        SwsContext*             ScaleContext;
+        AVFramePtr              ScaledFrame;
+        int                     Scale_OutputPixelFormat;
+        AVFramePtr              OutputFrame;
+
+
+        // FFmpeg pointers - Output
+        AVCodecContext*         Output_CodecContext;
+        int                     Output_PixelFormat;
+        int                     Output_CodecID;
+
+        // Out
+        outputmethod            OutputMethod;
+        Image                   image;
+
+        struct AVPacketDeleter {
+            void operator()(AVPacket* packet);
+        };
+
+        std::vector<std::shared_ptr<AVPacket>>  Thumbnails;
+        std::vector<AVFramePtr> Panels;
+
+        size_t                  Thumbnails_Modulo;
+        CommonStats*            Stats;
+
+        // Status
+        size_t                  FramePos;               // Current position of playback
+        double                  TimeStamp;              // Current position of playback
+
+        // Helpers
+        bool                    initEncoder(const QSize& size);;
+        bool                    FilterGraph_Init();
+        void                    FilterGraph_Free();
+        bool                    Scale_Init();
+        void                    Scale_Free();
+        bool                    AdaptDAR();
+        double                  GetDAR();
+
+        int                     Width;
+        int                     Height;
+        bool scaleBeforeEncoding = { false }; // used to change format before encoding
+
+        std::unique_ptr<AVPacket, outputdata::AVPacketDeleter> FFmpeg_Glue::outputdata::encodeFrame(AVFrame* frame, bool* ok = nullptr);
```

Comment:
On some compilers (Microsoft if conformance mode set [warning](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/c4596?view=vs-2019) becomes error, LLVM error), it leads to `error: extra qualification`, remove instances of `outputdata::` and  `FFmpeg_Glue::outputdata::`.

Reply:
same on my mac
```
../../../../Source/Core/FFmpeg_Glue.h:180:89: error: extra qualification on member 'encodeFrame'
        std::unique_ptr<AVPacket, outputdata::AVPacketDeleter> FFmpeg_Glue::outputdata::encodeFrame(AVFrame* frame, bool* ok = nullptr);
                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~^
```